### PR TITLE
chore: push json to private project

### PR DESCRIPTION
**Pull request by VebjornG**
### Description

This is a script that creates a json file with keys and values from the docstrings of the functions in InDSL and compares them to the source of truth in Locize. If any changes have been made they will be pushed to Locize, replacing the old ones. The script is used in the workflow `push-locize.yaml` which is run when translations are needed for widespread use of InDSL. We use e.g. `..._1_0` as versions for the versioned functions rather than `"..._1.0"` because Locize will treat any key name with a dot as an array an split the key name at the dot: https://github.com/locize/locize-cli/issues/23. An example of from the raw output from Locize with `_1.0` as the versioning number is:

`"INDSL_DETECT_OUTLIERS_1": {
    "0_": "Outlier detection"
  },`
  
  [Ticket](https://cognitedata.atlassian.net/browse/AH-2725?atlOrigin=eyJpIjoiMzU3YWVlNTMzMDViNGQ5MTgyYTNlZjk1OGZhNjEzZTEiLCJwIjoiaiJ9)

### CODE REVIEW

File: translations/create_translation_json.py [hunk 1]
```diff
@@ -18,6 +19,11 @@
 TOOLBOX_NAME = "TOOLBOX_NAME"
 COGNITE = "__cognite__"
 
+LOCIZE_API_KEY = "9e22ecf1-1fe7-41f2-827b-9f51b7c67f8b"
```
Comment (tuanng-cognite) [inline translations/create_translation_json.py:22]: it is not recommended to check in these values even if this is a test. Environment variable should be used instead

File: translations/create_translation_json.py [hunk 2]
```diff
@@ -125,9 +131,56 @@ def create_json_file():
...
+    pull_response = requests.get(pull_url, headers=headers, timeout=30)
+    if pull_response.status_code == 200:
+        with open(OUTPUT_FILE, "w") as file:
+            file.write(pull_response.text)
+    else:
+        pull_response.raise_for_status()
+
+    # Get the keys from locize, this is the source of truth
+    with open("keys_from_locize.json", "r") as file:
+        keys_from_locize = json.load(file)
```
Comment (tuanng-cognite) [inline translations/create_translation_json.py:156]: don't need to read from file. We can parse it from response `pull_response`

File: translations/create_translation_json.py [hunk 3]
```diff
@@ -125,9 +131,56 @@ def create_json_file():
...
+
+    # Get new keys and values with potential changes from the create_json_file() function
+    with open("translated_operations.json", "r") as file:
+        translated_operations = json.load(file)
```
Comment (tuanng-cognite) [inline translations/create_translation_json.py:160]: also don't need to load from file. `create_mapping_for_translations` should return that

**Comment by tuanng-cognite** [comment]
With that limitation, I think we should change the key generation to exclude `.` character

**Comment by VebjornG** [comment]
> With that limitation, I think we should change the key generation to exclude `.` character

That is done already: `version_str = f"_{_format_str(version.replace('.', '_'))}" if version else ""`

File: .github/workflows/push-locize.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: Push JSON to Locize
 
 on:
-  workflow_dispatch:
+  pull_request: {}
```
Comment (tuanng-cognite) [inline .github/workflows/push-locize.yaml:4]: should remove this change

File: translations/create_translation_json.py [hunk 5]
```diff
@@ -18,6 +20,10 @@
 TOOLBOX_NAME = "TOOLBOX_NAME"
 COGNITE = "__cognite__"
 
+LOCIZE_API_KEY = os.getenv("LOCIZE_API_KEY")
```
[the remark this answers is not in the document]
Comment (tuanng-cognite) [reply translations/create_translation_json.py:23]: the action actually mounts the secret as environment variables. So it should work and is tested

[the remark this answers is not in the document]
Comment (tuanng-cognite) [reply translations/create_translation_json.py:23]: https://github.com/cognitedata/indsl/pull/118/files#diff-af51498b0cecab5851b09aab759e6f9aedb9d220e57369e9dc1c7f0216c44092R30-R31

**Review by tuanng-cognite — APPROVED** [review]
small comment. Otherwise, look good to me

**Review by tuanng-cognite — APPROVED** [review]

**Pull request merged**